### ISSUE TITLE
'Class based routes' shift ignoring namespace method.

### ISSUE THREAD

**Issue by devNoiseConsulting** [laravel-shift/workbench-issues#46]
**Desktop:**
 - OS: Mac
 - OS Version: 10.14.6
 - App Version: 0.6.0 (0.6.0)

**Describe the bug**
Ran the 'Class based routes' shift on an app and ended up with incorrect paths to classes. Shift did not seem to take the namespace into consideration when creating class path. 

Default path in RouteServiceProvider was: App\\Http\\Controllers
Namespace method adds: Clients

**To Reproduce**
1. Create a route group using a namespace.
```
        // Clients
        Route::name('clients.')
            ->namespace('Clients')
            ->prefix('clients')
            ->group(function () {
                Route::get('/', 'ClientsController@index')->name('list');
                Route::post('/', 'EditController@create')->name('new');

                Route::get('create', 'EditController@index')->name('create');

                Route::get('/{id}', 'EditController@index')->name('edit')->where('id', '\d+');
                Route::put('/{id}', 'EditController@update')->name('update')->where('id', '\d+');
            });
```
2. Run 'Class based routes'
```
$ php artisan route:list --json

   Illuminate\Contracts\Container\BindingResolutionException

  Target class [App\Http\Controllers\UsersController] does not exist.

  at vendor/laravel/framework/src/Illuminate/Container/Container.php:875
    871▕
    872▕         try {
    873▕             $reflector = new ReflectionClass($concrete);
    874▕         } catch (ReflectionException $e) {
  ➜ 875▕             throw new BindingResolutionException("Target class [$concrete] does not exist.", 0, $e);
    876▕         }
    877▕
    878▕         // If the type is not instantiable, the developer is attempting to resolve
    879▕         // an abstract type such as an Interface or Abstract Class and there is
```
Result of shift:
```
use App\Http\Controllers\ClientsController;
use App\Http\Controllers\EditController;
use Illuminate\Support\Facades\Route;

        Route::name('clients.')
            ->prefix('clients')
            ->group(function () {
                Route::get('/', [ClientsController::class, 'index'])->name('list');
                Route::post('/', [EditController::class, 'create'])->name('new');

                Route::get('create', [EditController::class, 'index'])->name('create');

                Route::get('/{id}', [EditController::class, 'index'])->name('edit')->where('id', '\d+');
                Route::put('/{id}', [EditController::class, 'update'])->name('update')->where('id', '\d+');
            });
```

Expected result shift:
```
use App\Http\Controllers\Clients\ClientsController;
use App\Http\Controllers\Clients\EditController;
use Illuminate\Support\Facades\Route;

        Route::name('clients.')
            ->prefix('clients')
            ->group(function () {
                Route::get('/', [ClientsController::class, 'index'])->name('list');
                Route::post('/', [EditController::class, 'create'])->name('new');

                Route::get('create', [EditController::class, 'index'])->name('create');

                Route::get('/{id}', [EditController::class, 'index'])->name('edit')->where('id', '\d+');
                Route::put('/{id}', [EditController::class, 'update'])->name('update')->where('id', '\d+');
            });
```

**Comment by devNoiseConsulting** [comment]
This app has an EditController in the Application, Clients, and Users namespaces. I understand that the routes still might have some problems after the shift. It makes sense that I would have to add aliases to those imports.

```
use App\Http\Controllers\Applications\EditController as ApplicationsEditController;
use App\Http\Controllers\Clients\EditController as ClientsEditController;
use App\Http\Controllers\Users\EditController as UsersEditController;
```

**Comment by jasonmccreary** [comment]
@devNoiseConsulting, this case should be covered. Can you update to 0.6.1 and see if it still happens. If so, we'll run your snippet through our test suite to see what might be missing.

**Comment by devNoiseConsulting** [comment]
@jasonmccreary Updated workbench to 0.6.1. Docker image is v19. The result of the shift is the same as the one from the 0.6.0 version.

**Comment by jasonmccreary** [comment]
@devNoiseConsulting, shucks. I'll take your code snippet and run it through our test suite to see what it does.

**Comment by jasonmccreary** [comment]
@devNoiseConsulting, I understand more of what you are saying now. Did it import _any_ of these nested namespace controllers correctly. Or were they all simply top level?

**Comment by devNoiseConsulting** [comment]
@jasonmccreary None of the nested namespaces made it into the class path for the controllers. Also found another example in my api.php where it used the namespace method twice and neither of those made it into the class path.


Starting route:
```
Route::middleware('auth:api')
    // not sure why I had to prefix with 'api.' here.
    // Is should have been done in the RouteServiceProvider
    ->name('api.v1.')
    ->namespace('API\v1')
    ->prefix('v1')
    ->group(function () {
        Route::name('users.')
            ->namespace('Users')
            ->prefix('users')
            ->group(function () {
                Route::get('/{id}', 'UsersController@byId')->name('byId')->where('id', '\d+');
                Route::post('/token', 'UsersController@byToken')->name('byToken');
                Route::patch('{id}/token', 'UsersController@setToken')->name('setToken')->where('id', '\d+');
                Route::post('/credentials', 'UsersController@byCredentials')->name('byCredentials');
            });
    });
```

Result of shift:
```
use App\Http\Controllers\UsersController;

Route::middleware('auth:api')
    // not sure why I had to prefix with 'api.' here.
    // Is should have been done in the RouteServiceProvider
    ->name('api.v1.')
    ->prefix('v1')
    ->group(function () {
        Route::name('users.')
            ->prefix('users')
            ->group(function () {
                Route::get('/{id}', [UsersController::class, 'byId'])->name('byId')->where('id', '\d+');
                Route::post('/token', [UsersController::class, 'byToken'])->name('byToken');
                Route::patch('{id}/token', [UsersController::class, 'setToken'])->name('setToken')->where('id', '\d+');
                Route::post('/credentials', [UsersController::class, 'byCredentials'])->name('byCredentials');
            });
    });
```

Expected result shift:
```
use App\Http\Controllers\API\v1\Users\UsersController;

Route::middleware('auth:api')
    // not sure why I had to prefix with 'api.' here.
    // Is should have been done in the RouteServiceProvider
    ->name('api.v1.')
    ->prefix('v1')
    ->group(function () {
        Route::name('users.')
            ->prefix('users')
            ->group(function () {
                Route::get('/{id}', [UsersController::class, 'byId'])->name('byId')->where('id', '\d+');
                Route::post('/token', [UsersController::class, 'byToken'])->name('byToken');
                Route::patch('{id}/token', [UsersController::class, 'setToken'])->name('setToken')->where('id', '\d+');
                Route::post('/credentials', [UsersController::class, 'byCredentials'])->name('byCredentials');
            });
   });
```

**Comment by jasonmccreary** [comment]
Thanks for the additional code snippets. We'll get these added to our test cases and patch up in the next release.

**Comment by jasonmccreary** [comment]
The should be fixed as of version `0.7.5`. While Shift still may not convert all the example code exactly as you have, it should do a better job with nested namespaces.
